Remotion/ObjectBinding/Core/BindableObject/InterfaceImplementationPropertyInformation.cs

```diff
@@ -119,22 +119,30 @@ public bool CanBeSetFromOutside
       get { return GetSetMethod (false) != null; }
     }
 
-    public object GetValue (object instance, object[] indexParameters)
+    public object? GetValue (object? instance, object?[]? indexParameters)
     {
-      ArgumentUtility.CheckNotNull ("instance", instance);
+      ArgumentUtility.CheckNotNull ("instance", instance!);
 
-      return GetGetMethod (true).Invoke (instance, indexParameters);
+      var getMethod = GetGetMethod (true);
+
+      if (getMethod == null)
+        throw new ArgumentException ("The get accessor could not be found.");
+
+      return getMethod.Invoke (instance, indexParameters);
     }
 
-    public void SetValue (object instance, object value, object[] indexParameters)
+    public void SetValue (object? instance, object? value, object?[]? indexParameters)
     {
-      ArgumentUtility.CheckNotNull ("instance", instance);
+      ArgumentUtility.CheckNotNull ("instance", instance!);
 
       var setMethod = GetSetMethod (true);
 
+      if (setMethod == null)
+        throw new ArgumentException ("The set accessor could not be found.");
```

Comment:
@MichaelKetting these Exceptions are of the wrong Type, with the wrong message - what should I do instead?
```csharp
new InvalidOperationException ($"{_implementationPropertyInfo.Name} does not declare a getter.");
```
or setter, respectively?

Reply:
+ enhance docs of course